Slam/DefaultAddMatchInteractor.swift

```diff
@@ -2,12 +2,20 @@ import Foundation
 
 public class DefaultAddMatchInteractor : AddMatchInteractor {
 
+    var api : SlamAPI?
     let view : AddMatchView
+
     public init(view: AddMatchView) {
         self.view = view
     }
+    
+    public init(view: AddMatchView, api: SlamAPI) {
+        self.view = view
+        self.api = api
+    }
 
     public func attemptAdd(#playerOne: String, playerTwo: String) {
+        api!.addMatch(playerOne, playerTwo: playerTwo)
         view.matchWasAdded()
```

Comment:
This should be happening in a callback that is passed to `addMatch`. Right now, we are always telling `view` that the match was successfully added, even when the call fails.